crates/wasm-mutate/tests/tests.rs

```diff
@@ -0,0 +1,98 @@
+
+
+#[cfg(test)]
+mod tests{
+
+    use wasm_encoder::Instruction;
+    use wasm_mutate::WasmMutate;
+    use wasmparser::{Chunk, Operator, Parser, Payload, Validator};
+
+
+    // Copied from wasm-smith
+    fn validate(validator: &mut Validator, bytes: &[u8]) {
+        let err = match validator.validate_all(bytes) {
+            Ok(()) => return,
+            Err(e) => e,
+        };
+        drop(std::fs::write("test.wasm", &bytes));
+        if let Ok(text) = wasmprinter::print_bytes(bytes) {
+            drop(std::fs::write("test.wat", &text));
+        }
+
+        panic!("wasm failed to validate {:?}", err);
+    }
+
+    /// Since there is no code section, any mutator registered under that pattern wont affect the module
+    #[test]
+    fn idempotent_header() {
+        let original = b"\0asm\x01\0\0\0";
+        let mutator = WasmMutate::default();
+        
+        let mutated = mutator.run(original).unwrap();
+
+        assert_eq!(original.to_vec(), mutated)
+    }
+
+
+    /// Since there is no code section, any mutator registered under that pattern wont affect the module
+    #[test]
+    fn test_code_mutator() {
+        // From https://developer.mozilla.org/en-US/docs/WebAssembly/Text_format_to_wasm
+        let wat = r#"
+        (module
+            (func (result i64)
+                i64.const 42
+            )
+        )
+        "#;
+        let original = &wat::parse_str(wat).unwrap();
+        let mutator = WasmMutate::default();
+        // seed is zero, which means first mutator
```

Comment:
This is going to make the tests fragile. If we reorder the mutators then we will break tests.

For testing specific mutators, it may make more sense to have "unit-test" style tests inside the module that defines the mutator (with `#[cfg(test)] mod tests { ... }`) than to have "integration-test" style tests in `tests/*`. This is because a submodule inside the same module that defines the mutator has access to the mutator directly because of the scope it is in even if the mutator isn't publicly exported, and we don't have rely on RNG seeds and mutator ordering.

Reply:
Addressed in https://github.com/bytecodealliance/wasm-tools/pull/336